viadot/flows/adls_to_azure_sql.py

```diff
@@ -113,15 +189,21 @@ def check_dtypes_sort(
                 new_dtypes = dict()
                 for key in df.columns:
                     new_dtypes.update([(key, dtypes[key])])
+
+                check_hardcoded_dtypes_len(df, new_dtypes)
+
             else:
                 new_dtypes = dtypes.copy()
+                check_hardcoded_dtypes_len(df, new_dtypes)
+
         else:
             logger.error("There is a discrepancy with any of the columns.")
             raise signals.FAIL(
                 "dtype dictionary contains key(s) that not matching with the ADLS file columns name, or they have different length."
             )
     else:
         new_dtypes = dtypes.copy()
+        check_hardcoded_dtypes_len(df, new_dtypes)
```

Comment:
There is no need to put the same function in 3 places. Let's call this function right above the `return` because there we will have the final `new_dtypes`